osbs/api.py

```diff
@@ -127,15 +129,58 @@ def create_build_from_buildrequest(self, build_request, namespace=DEFAULT_NAMESP
         build_response = BuildResponse(response)
         return build_response
 
+    def _create_build_config_and_build(self, build_json, namespace):
+        # TODO: test this method more thoroughly
+        build_config_name = build_json['metadata']['name']
+
+        # check if a build already exists for this config; if so then raise
+        # TODO: could we do this more effectively than traversing all?
+        running_builds = self.os.list_builds(namespace=namespace).json()
+        for rb in running_builds['items']:
+            try:
+                phase = rb['status']['phase'].lower()
```

Comment:
I would prefer using one piece of code for this kind of stuff: https://github.com/projectatomic/osbs-client/blob/c60ede765537d322135431ecda241d88d045dab5/osbs/build/build_response.py#L51